### ISSUE TITLE
[Bug]: resource not found when creating load balancer services

### ISSUE THREAD

**Issue by nhedger** [exoscale/terraform-provider-exoscale#380]
### Current Behavior

When provisioning a load balancer and its services, I get the following error:

```
Error: Post "https://api-ch-gva-2.exoscale.com/v2/load-balancer/03062200-483e-42a8-a375-d38b7d77adc7/service": resource not found

  with module.network.exoscale_nlb_service.http,
  on modules/network/network-load-balancer-service.tf line 5, in resource "exoscale_nlb_service" "http":
   5: resource "exoscale_nlb_service" "http" {
```

This error message suggests that load-balancer with ID `03062200-483e-42a8-a375-d38b7d77adc7` cannot be found, but it has been creating just before.

This does not happen all the time, I first provisioned with success, then deleted all created resources and when I ran again I got the error. I'm wondering if there's some sort of race condition.

### Expected Behavior

The load balancer services are correctly created.

### Steps To Reproduce

Create a network load balancer

```terraform
resource "exoscale_nlb" "load-balancer" {
  zone = "ch-gva-2"
  name = "staging"
}
```

Create a service

```terraform
resource "exoscale_nlb_service" "http" {
  nlb_id = exoscale_nlb.load-balancer.id
  zone   = exoscale_nlb.load-balancer.zone
  name   = "http"

  instance_pool_id = var.nodepool.id
  protocol         = "tcp"
  port             = 80
  target_port      = 3080
  strategy         = "round-robin"

  healthcheck {
    mode     = "http"
    port     = 30080
    uri      = "/healthz"
    interval = 5
    timeout  = 3
    retries  = 1
  }
}
```

### Provider Version

0.60.0

### Terraform Version

OpenTofu v1.7.3

### Relevant log output

```shell
module.network.exoscale_nlb_service.sftp: Creating...
module.network.exoscale_nlb_service.ftp: Creating...
module.network.exoscale_nlb_service.http: Creating...
module.network.exoscale_nlb_service.https: Creating...

Error: Post "https://api-ch-gva-2.exoscale.com/v2/load-balancer/03062200-483e-42a8-a375-d38b7d77adc7/service": resource not found

  with module.network.exoscale_nlb_service.http,
  on modules/network/network-load-balancer-service.tf line 5, in resource "exoscale_nlb_service" "http":
   5: resource "exoscale_nlb_service" "http" {


Error: Post "https://api-ch-gva-2.exoscale.com/v2/load-balancer/03062200-483e-42a8-a375-d38b7d77adc7/service": resource not found

  with module.network.exoscale_nlb_service.https,
  on modules/network/network-load-balancer-service.tf line 26, in resource "exoscale_nlb_service" "https":
  26: resource "exoscale_nlb_service" "https" {


Error: Post "https://api-ch-gva-2.exoscale.com/v2/load-balancer/03062200-483e-42a8-a375-d38b7d77adc7/service": resource not found

  with module.network.exoscale_nlb_service.ftp,
  on modules/network/network-load-balancer-service.tf line 47, in resource "exoscale_nlb_service" "ftp":
  47: resource "exoscale_nlb_service" "ftp" {


Error: Post "https://api-ch-gva-2.exoscale.com/v2/load-balancer/03062200-483e-42a8-a375-d38b7d77adc7/service": resource not found

  with module.network.exoscale_nlb_service.sftp,
  on modules/network/network-load-balancer-service.tf line 67, in resource "exoscale_nlb_service" "sftp":
  67: resource "exoscale_nlb_service" "sftp" {

Error: OpenTofu exited with code 1.
Error: Process completed with exit code 1.
```

**Comment by kobajagi** [comment]
Thanks @nhedger , it does look like a race condition.
It's probably not a terraform provider issue but rather API problem. I'll look into it.

**Comment by nhedger** [comment]
Hey @kobajagi, after doing some more digging into this, I believe this error is on my side.

I was referencing the id of the **nodepool**, where I believe I should have referenced the id of the **instance pool** that belongs to the **nodepool**. I think I got confused between the two, and now that I reference it like following, the creation seems to work:

```terraform
instance_pool_id = var.nodepool.instance_pool_id
```

When I saw the 404, I assumed it was for the top resource (load balancer), but it seems to have been for the instance pool. The API was merely trying to find a instance pool using a nodepool id, which of course doesn't match.

I apologize for wasting your time on this.

Cheers
